assets/js/lock-init.js

```diff
@@ -25,6 +25,11 @@ jQuery(document).ready(function ($) {
         Cookies.set( opts.nonceCookieName, opts.settings.auth.params.nonce );
     }
 
+    // Look for additional fields to display
+    if ( wpAuth0LockGlobalFields ) {
+        opts.settings.additionalSignUpFields = wpAuth0LockGlobalFields;
```

Comment:
Q: How does this setup work?

Reply:
The global `wpAuth0LockGlobalFields` variable is set earlier in execution (the other file you're reviewing) and this looks to see whether there is anything there and assigns if so. Previously, the block that set the fields JS was trying to assign to the `wpAuth0LockGlobal` and causing problems in some setups (the global not being available). JS globals are not the greatest but they're commonly used in WP to pass information from PHP or the DB into JS. These fields are stored as JS, not JSON, so I can't parse, store, then output like I would like to.